packages/components/src/navigation/use-navigation-tree-nodes.tsx

```diff
@@ -3,25 +3,29 @@
  */
 import { useState } from '@wordpress/element';
 
-export const useNavigationTreeNodes = () => {
-	const [ nodes, setNodes ] = useState( {} );
+export function useNavigationTreeNodes<
+	TNode extends { children?: React.ReactNode; [ key: string ]: unknown }
+>() {
+	const [ nodes, setNodes ] = useState<
+		Record< string, Omit< TNode, 'children' > >
```

Comment:
The omission of the `children` prop is caused by the `const { children, ...newNode } = value;` prop destructuring a few lines below in the `addNode` function.